### PR TITLE
Implementation of "k-decay Method For Learning Rate Schedule"

### PR DESCRIPTION
**Pull request by atztao**
See [Tao Zhang & Wei Li, 2020], k-decay: A New Method For Learning Rate Schedule. 
https://arxiv.org/abs/2004.05909

> This paper,  we propose the k-decay method, in which the LR is changed by its k-th order derivative, to obtain the new LR schedule. In the new LR schedule, a hyper-parameter $k$ is used to control the change degree of LR, whereas the
  original method of $k$ at 1. We then derive the k-decay factors $\frac{t^k}{T^k}$ for LR schedule and apply it to polynomial, cosine and exponential function. By repeatedly using the k-decay method, one can identify the best LR schedule. We then evaluate the k-decay method on CIFAR And ImageNet datasets with different neural networks (ResNet, Wide ResNet and DenseNet). Our experiments show that the k-decay method can achieve improvements over the state-of-the-art results on most of them. The accuracy can be improved by 1.08 % on the CIFAR-10 dataset, and by 2.07 % on the CIFAR-100 dataset. On the ImageNet, the accuracy can be improved by 1.25%.  Our method is not only efficient in computation but easy to use, and thus greatly improve the training performance.

image

### CODE REVIEW

**Comment by atztao** [comment]
@googlebot I signed it!

**Comment by atztao** [comment]
@omalleyt12  Could you please review this PR? Thank you!

**Comment by atztao** [comment]
@gbaned It seems like @omalleyt12 is currently busy with other things or doesn't receive GitHub notifications. Could you request a review from another member of the TensorFlow team?

**Comment by gbaned** [comment]
@tanzheny Can you please resolve conflicts? Thanks!

**Comment by atztao** [comment]
> @atztao Can you please resolve conflicts? Thanks!

Thanks for your remainding me.

**Comment by atztao** [comment]
> @atztao Can you please resolve conflicts? Thanks!

Thanks for your remainding me.

**Comment by atztao** [comment]
@tanzhenyu  Could you please review this PR? This is just a simple and small change. It has been submitted for more than ten days. Thanks!

**Comment by atztao** [comment]
@gbaned Sorry to bother you again. But this PR has been submitted for more than ten days. It seems like @tanzhenyu  is also busy with other things and doesn't receive GitHub notifications. Could you request a review from another member of the TensorFlow team? Thanks!

**Comment by omalleyt12** [comment]
@atztao Thanks for the review! I think the best place for a new object like this is in the [TensorFlow Addons Repo](https://github.com/tensorflow/addons)

Could you please open a PR there? Thank you!

**Comment by atztao** [comment]
@omalleyt12 Thanks for your advice. But you can see this method in the PR is not a new fuction, it just perfect the learning rate decay. I only changed $\frac{t}{T}$ to $\frac{t^k}{T^k}$. And i found the SGDR method in the core, that's good schedule, but this method is more better. You can see the experiments part. Please think about it again.

**Comment by atztao** [comment]
@gbaned Sorry to bother you again. Please request a review from another member of the TensorFlow team. Thanks!

**Comment by gbaned** [comment]
> @gbaned Sorry to bother you again. Please request a review from another member of the TensorFlow team. Thanks!

@atztao Sorry for the delay. We are discussing about this PR internally. Thank you!

**Comment by mihaimaruseac** [comment]
We cannot integrate code from all research papers. Instead, new code should go into [Addons Repository](https://github.com/tensorflow/addons) and, in time, after careful experiments and seeing that it properly works on all use cases we can incorporate the change and make it default. Until then, the new schedule has to be provided via an optional layer/operator/code path.
